astra/src/test/java/com/slack/astra/graphApi/GraphBuilderTest.java

```diff
@@ -0,0 +1,610 @@
+package com.slack.astra.graphApi;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.slack.astra.zipkinApi.ZipkinEndpointResponse;
+import com.slack.astra.zipkinApi.ZipkinSpanResponse;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class GraphBuilderTest {
+  private GraphBuilder defaultGraphBuilder;
+  private GraphBuilder configuredGraphBuilder;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    defaultGraphBuilder = new GraphBuilder(GraphConfig.DEFAULT);
+
+    // Load custom config from YAML file
+    Path configPath =
+        new File(
+                getClass()
+                    .getClassLoader()
+                    .getResource("test-dependency-graph-config.yaml")
+                    .getFile())
+            .toPath();
+    GraphConfig customConfig = GraphConfig.load(configPath);
+    configuredGraphBuilder = new GraphBuilder(customConfig);
+  }
+
+  @Test
+  void buildFromSpans_emptyList_returnsEmptyGraph() {
+    List<ZipkinSpanResponse> spans = new ArrayList<>();
+    Graph graph = defaultGraphBuilder.buildFromSpans(spans);
+
+    assertThat(graph.nodes).isEmpty();
+    assertThat(graph.edges).isEmpty();
+  }
+
+  @Test
+  void buildFromSpans_singleSpanWithoutParent_createsSingleNodeWithoutEdges() {
+    List<ZipkinSpanResponse> spans = new ArrayList<>();
+    ZipkinSpanResponse span =
+        createSpanWithTags(
+            "span1",
+            "trace1",
+            null,
+            Map.of(
+                "kube.app",
+                "app1",
+                "kube.namespace",
+                "ns1",
+                "kube.operation",
+                "op1",
+                "resource",
+                "res1"));
+    spans.add(span);
```

Comment:
I'd probably use List.of() here, to remove some incidental complexity.
```suggestion
    List<ZipkinSpanResponse> spans = List.of(
        createSpanWithTags(
            "span1",
            "trace1",
            null,
            Map.of(
                "kube.app",
                "app1",
                "kube.namespace",
                "ns1",
                "kube.operation",
                "op1",
                "resource",
                "res1")));
```